qa/tasks/cephfs/test_exports.py

```diff
@@ -114,6 +117,8 @@ def test_export_pin(self):
             self._wait_subtrees(status, 0, [('/1', 0), ('/1/4/5', 1), ('/a', 1), ('/a/b/aa/bb', 0)])
 
         # Test getfattr
+	if (str(subprocess.check_output(['uname', '-r'])) < "5" and isinstance(self.mount_a, KernelMount) and not(self.mount_a.getfattr("1", "ceph.dir.pin"))):
```

Comment:
You're running `uname -r` on the machine executing this Python script which is not the same machine as what's running the kernel mount.

What I would recommend is consider doing the following: use try/catch to execute the getfattr command. If it fails and the reason is `ENODATA`, then you should check that the mount is a (a) a kernel mount and (b) older than v5.0 of the linux kernel. If it is, then we can skip the test. Otherwise, something else is wrong and the error should be raised to abort the test.

See for example on how to run `uname -r` on the kernel: https://github.com/ceph/ceph/blob/f7786d6cf5483dc874ea9f581a81c67d194a3b12/qa/tasks/cephfs/test_journal_repair.py#L108-L127